libraries-server-2/src/main/java/com/baeldung/netty/customhandlersandlisteners/ChatServerMain.java

```diff
@@ -0,0 +1,52 @@
+package com.baeldung.netty.customhandlersandlisteners;
+
+import com.baeldung.netty.customhandlersandlisteners.handler.ServerEventHandler;
+import com.baeldung.netty.customhandlersandlisteners.listener.ChannelInfoListener;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.handler.codec.string.StringDecoder;
+import io.netty.handler.codec.string.StringEncoder;
+
+public final class ChatServerMain {
+
+    public static final String HOST = "localhost";
+    public static final int PORT = 8081;
+
+    public static void main(String[] args) {
+        EventLoopGroup serverGroup = new NioEventLoopGroup(1);
+        EventLoopGroup clientGroup = new NioEventLoopGroup();
+        try {
```

Comment:
We can simplify the try with a more updated version, that is, with try-with-resources.
```suggestion
    try (EventLoopGroup clientGroup = new NioEventLoopGroup(); EventLoopGroup serverGroup = new NioEventLoopGroup(1)) {
```

We would remove the finally part too.

Reply:
This one would result in error "The resource type EventLoopGroup does not implement java.lang.AutoCloseable".